From: Marc Zyngier <marc.zyngier@arm.com>
Date: Tue, 7 Nov 2017 10:04:38 +0000
Subject: irqchip/gic-v3-its: Fix VPE activate callback return value
Patch-mainline: v4.15-rc1
Git-commit: 6ef930f20c30f8a7dcffa50fa9f33a9211727a6e
References: FATE#323954

its_vpe_irq_domain_activate should always return 0. Really. There
is not a single case why it wouldn't.  So this "return true;" is
really a copy/paste issue that got revealed now that we actually
check the return value of the activate method.

Brown paper bag day.

Fixes: 2247e1bf7063 ("irqchip/gic-v3-its: Limit scope of VPE mapping to be per ITS")
Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
Acked-by: Alexander Graf <agraf@suse.de>
---
 drivers/irqchip/irq-gic-v3-its.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -2826,7 +2826,7 @@
 
 	/* If we use the list map, we issue VMAPP on demand... */
 	if (its_list_map)
-		return true;
+		return 0;
 
 	/* Map the VPE to the first possible CPU */
 	vpe->col_idx = cpumask_first(cpu_online_mask);
